src/windows/functions.cpp

```diff
@@ -393,6 +393,38 @@ MOUNTUTILS_RESULT EjectDriveLetter(TCHAR driveLetter) {
   return MOUNTUTILS_SUCCESS;
 }
 
+BOOL IsDriveLetterEjectable(TCHAR driveLetter) {
+  TCHAR devicePath[8];
+  wsprintf(devicePath, TEXT("%c:\\"), driveLetter);
+
+  MountUtilsLog("Checking whether drive is ejectable: "
+      + std::string(1, driveLetter));
+
+  switch (GetDriveType(devicePath)) {
+    case DRIVE_NO_ROOT_DIR:
+      MountUtilsLog("The drive letter doesn't exist");
+      return FALSE;
+    case DRIVE_REMOVABLE:
+      MountUtilsLog("The drive letter is removable");
+      return TRUE;
+    case DRIVE_FIXED:
+      MountUtilsLog("The drive letter is fixed");
+      return TRUE;
+    case DRIVE_REMOTE:
+      MountUtilsLog("The drive letter is remote");
+      return FALSE;
+    case DRIVE_CDROM:
+      MountUtilsLog("The drive letter is a CDROM");
+      return FALSE;
+    case DRIVE_RAMDISK:
+      MountUtilsLog("The drive letter is a RAM disk");
+      return FALSE;
+    default:
+      MountUtilsLog("The drive letter type is unknown");
```

Comment:
I'd suggest to change "drive letter" -> "drive", as the drive letter is just a label for the volume, not the the physical device (and also shorter)